Source/Eto.WinForms/Forms/Controls/TreeViewHandler.cs

```diff
@@ -15,12 +15,30 @@ public class TreeViewHandler : WindowsControl<swf.TreeView, TreeView, TreeView.I
 		static readonly string EmptyName = Guid.NewGuid().ToString();
 		bool ignoreExpandCollapseEvents = true;
 
-		public TreeViewHandler()
+		// flicker-free with no effort, works since Vista, but won't hurt on XP
+		class OptimizedTreeView : swf.TreeView
 		{
-			this.Control = new swf.TreeView
+			public OptimizedTreeView()
 			{
-				HideSelection = false
-			};
+				HideSelection = false;
+				SetStyle
+					( swf.ControlStyles.OptimizedDoubleBuffer
+					| swf.ControlStyles.AllPaintingInWmPaint
+					, true);
+			}
+
+			const int TVS_EX_DOUBLEBUFFER = 0x0004;
+			protected override void OnHandleCreated(EventArgs e)
+			{
+				base.OnHandleCreated(e);
+				Win32.SendMessage(Handle, Win32.WM.TVM_SETEXTENDEDSTYLE,
+					(IntPtr)TVS_EX_DOUBLEBUFFER, (IntPtr)TVS_EX_DOUBLEBUFFER);
```

Comment:
Mono Windows.Forms would never work without adding `SendMessage`, this is acceptable P/Invoke (and I did not add any, just used what was there already)